lib/services/API/country_educational_institute_api.dart

```diff
@@ -0,0 +1,49 @@
+import 'package:mobile_app/constants.dart';
+import 'package:mobile_app/models/failure_model.dart';
+import 'package:mobile_app/utils/api_utils.dart';
+
+abstract class CountryInstituteAPI {
+  Future<List<String>> getSuggestions(String query, String url);
+}
+
+class HttpCountryInstituteAPI implements CountryInstituteAPI {
+  bool _countryFlag = true;
+  bool _instituteFlag = true;
```

Comment:
I think you only need 1 variable since we need to check if its either country flag or institute flag.

Reply:
I guess you don't even need the flag since you can use the constant to check.

Reply:
The flags ensure that API call is made only once. If I don't use the flags then on every pattern change , a call is made to API. This is the purpose of flags.

Reply:
I am not able to understand how the API call will be made only once ? For every change in pattern, the API has to be called every time to get the relevant matches

Reply:
So, helper list stores all the strings fetched from API, and all that `TypeAheadFormField` needs is a list of Strings(Suggestions). Maybe adding a print statement inside the `if` condition can make things more clearer.

Reply:
Can you again check my question ?
The API of countries/institutes will return matches using the query given.
Now the question is that the query changes everytime when user is trying to search the country/institute, then how is it possible to cache ?
Therefore, how you can call the API once when the pattern is changing everytime ?